toolkits/slack/arcade_slack/tools/chat.py

```diff
@@ -72,43 +92,508 @@ def send_dm_to_user(
         ],
     )
 )
-def send_message_to_channel(
+async def send_message_to_channel(
     context: ToolContext,
-    channel_name: Annotated[
-        str,
-        "The Slack channel name where you want to send the message. "
-        "Slack channel names are ALWAYS lowercase.",
-    ],
+    channel_name: Annotated[str, "The Slack channel name where you want to send the message. "],
     message: Annotated[str, "The message you want to send"],
 ) -> Annotated[dict, "The response from the Slack API"]:
     """Send a message to a channel in Slack."""
 
-    slackClient = WebClient(
-        token=context.authorization.token
-        if context.authorization and context.authorization.token
-        else ""
+    try:
+        slackClient = AsyncWebClient(
+            token=context.authorization.token
+            if context.authorization and context.authorization.token
+            else ""
+        )
+
+        # Step 1: Retrieve the list of channels
+        channels_response = await slackClient.conversations_list()
+        channel_id = None
+        for channel in channels_response["channels"]:
+            if channel["name"].lower() == channel_name.lower():
+                channel_id = channel["id"]
+                break
+
+        if not channel_id:
+            raise RetryableToolError(
+                "Channel not found",
+                developer_message=f"Channel with name '{channel_name}' not found.",
+                additional_prompt_content=format_conversations_as_csv(channels_response),
+                retry_after_ms=500,  # Play nice with Slack API rate limits
+            )
+
+        # Step 2: Send the message to the channel
+        response = await slackClient.chat_postMessage(channel=channel_id, text=message)
+
+    except SlackApiError as e:
+        error_message = e.response["error"] if "error" in e.response else str(e)
+        raise ToolExecutionError(
+            "Error sending message",
+            developer_message=f"Slack API Error: {error_message}",
+        )
+    else:
+        return {"response": response.data}
+
+
+@tool(
+    requires_auth=Slack(
+        scopes=["channels:read", "groups:read", "im:read", "mpim:read"],
     )
+)
+async def get_members_from_conversation_by_id(
+    context: ToolContext,
+    conversation_id: Annotated[str, "The ID of the conversation to get members for"],
+    limit: Annotated[
+        Optional[int], "The maximum number of members to return."
+    ] = MAX_PAGINATION_SIZE_LIMIT,
```

Comment:
If I want to "list all members in the conversation that has ID 12345", then wouldn't defaulting to a limited number MAX_PAGINATION_SIZE_LIMIT prevent the tool from getting all of the members if there were more than 200 in the conversation?

I wonder if defaulting to `-1` and treating that as infinite (don't stop until next_cursor is None) would be better.

Reply:
modified all tools to set limit to `None` by default, and refactored `async_paginate` accordingly